examples/ubuntu/templates/user-data.sh

```diff
@@ -15,59 +19,54 @@ DEBIAN_FRONTEND=noninteractive apt-get install -y \
     build-essential \
     unzip
 
-USER_NAME=runners
-useradd -m -s /bin/bash $USER_NAME
-USER_ID=$(id -ru $USER_NAME)
+USER_NAME=${runner_run_as}
+useradd -m -s /bin/bash "$USER_NAME"
+USER_ID=$(id -ru "$USER_NAME")
+
+# uncomment following line to be able to debug and login as the user via ssm
+# echo -e "test1234\ntest1234" | passwd $USER_NAME
 
 # install and configure cloudwatch logging agent
 wget https://s3.amazonaws.com/amazoncloudwatch-agent/ubuntu/amd64/latest/amazon-cloudwatch-agent.deb
 dpkg -i -E ./amazon-cloudwatch-agent.deb
 amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -s -c ssm:${ssm_key_cloudwatch_agent_config}
 
-# configure systemd for running service in users accounts
-cat >/etc/systemd/user@UID.service <<-EOF
-
-[Unit]
-Description=User Manager for UID %i
-After=user-runtime-dir@%i.service
-Wants=user-runtime-dir@%i.service
-
-[Service]
-LimitNOFILE=infinity
-LimitNPROC=infinity
-User=%i
-PAMName=systemd-user
-Type=notify
-
-[Install]
-WantedBy=default.target
-
-EOF
-
-echo export XDG_RUNTIME_DIR=/run/user/$USER_ID >>/home/$USER_NAME/.profile
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
+apt-get update
+DEBIAN_FRONTEND=noninteractive apt-get install -y docker-ce docker-ce-cli containerd.io docker-ce-rootless-extras
 
-systemctl daemon-reload
-systemctl enable user@UID.service
-systemctl start user@UID.service
+{
+  echo
+  echo "export XDG_RUNTIME_DIR=/run/user/$USER_ID"
+  echo "export DOCKER_HOST=unix:///run/user/$USER_ID/docker.sock"
+  echo "export PATH=/home/$USER_NAME/bin:$PATH"
+  echo
+} >> "/home/$USER_NAME/.profile"
 
-curl -fsSL https://get.docker.com/rootless >>/opt/rootless.sh && chmod 755 /opt/rootless.sh
-su -l $USER_NAME -c /opt/rootless.sh
-echo export DOCKER_HOST=unix:///run/user/$USER_ID/docker.sock >>/home/$USER_NAME/.profile
-echo export PATH=/home/$USER_NAME/bin:$PATH >>/home/$USER_NAME/.profile
+systemctl stop docker
+systemctl disable docker
 
 # Run docker service by default
-loginctl enable-linger $USER_NAME
-su -l $USER_NAME -c "systemctl --user enable docker"
+loginctl enable-linger "$USER_NAME"
+su -l "$USER_NAME" -c "systemctl --user enable docker"
```

Comment:
Docker service was not started, by using the docker rootless setup it got started by default.
```suggestion
su -l "$USER_NAME" -c "dockerd-rootless-setuptool.sh install"
```